pkg/ensurance/collector/cadvisor/cadvisor_linux.go

```diff
@@ -64,35 +77,33 @@ func NewCadvisor(podLister corelisters.PodLister) *CadvisorCollector {
 		return nil
 	}
 
-	c := CadvisorCollector{
-		Manager:   m,
-		podLister: podLister,
-	}
-
-	if err := c.Manager.Start(); err != nil {
+	if err := m.Start(); err != nil {
 		klog.Errorf("Failed to start cadvisor manager: %v", err)
 		return nil
 	}
 
-	return &c
+	return &CadvisorManager{
+		m,
+	}
 }
 
 // Stop cadvisor and clear existing factory
 func (c *CadvisorCollector) Stop() error {
-	if err := c.Manager.Stop(); err != nil {
-		return err
-	}
-
-	// clear existing factory
-	cadvisorcontainer.ClearContainerHandlerFactories()
-
 	return nil
 }
 
 func (c *CadvisorCollector) GetType() types.CollectType {
 	return types.CadvisorCollectorType
 }
 
+func (c *CadvisorManager) ContainerInfo(containerName string, query *info.ContainerInfoRequest) (*info.ContainerInfo, error) {
+	return c.GetContainerInfo(containerName, query)
+}
+
+func (c *CadvisorManager) ContainerInfoV2(containerName string, options cadvisorapiv2.RequestOptions) (map[string]cadvisorapiv2.ContainerInfo, error) {
```

Comment:
this function is not needed?

Reply:
<img width="778" alt="image" src="https://user-images.githubusercontent.com/15925403/158940212-65fbc7ed-9db1-48e9-9f45-6300609dd141.png">

Reply:
You can use GetContainerInfo and  GetContainerInfoV2, it no needs wrap functions